Index: src/main/java/com/blazemeter/jmeter/utils/TestInfoProcessor.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/blazemeter/jmeter/utils/TestInfoProcessor.java	(revision 70a83efbc4a56fa8bd3d5ec96d7f2ee039f7cbb0)
+++ src/main/java/com/blazemeter/jmeter/utils/TestInfoProcessor.java	(revision )
@@ -39,12 +39,15 @@
                                 responseOptions.getInt(JsonFields.OVERRIDE_RAMP_UP),
                                 responseOptions.getInt(JsonFields.OVERRIDE_THREADS));
                         testInfo.setOverrides(overrides);
+                        BmLog.debug("\n\nJsonObject from body: " + "\n\n\n" + jsonObject.toString() + "\n\n\n");
+
                     }
                 }
             }
 
         } catch (JSONException je) {
             BmLog.error("Error while creating TestInfo from JSON: " + je + "\n" + jsonObject.toString());
+            BmLog.error("JsonObject from body: " + "\n\n\n" + jsonObject.toString() + "\n\n\n");
         } catch (ClassCastException cce) {
             BmLog.error("Error while creating TestInfo from JSON: " + cce + "\n" + jsonObject.toString());
         }
