Index: src/main/java/com/blazemeter/jmeter/testexecutor/TestPanelGui.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/blazemeter/jmeter/testexecutor/TestPanelGui.java	(revision 152b310fddd1e79f076cd3ed7715862a262a0c1d)
+++ src/main/java/com/blazemeter/jmeter/testexecutor/TestPanelGui.java	(revision )
@@ -1,6 +1,7 @@
 package com.blazemeter.jmeter.testexecutor;
 
 import com.blazemeter.jmeter.testinfo.TestInfo;
+import com.blazemeter.jmeter.testinfo.TestInfoController;
 import com.blazemeter.jmeter.testinfo.TestInfoReader;
 import com.blazemeter.jmeter.testinfo.UserInfo;
 import com.blazemeter.jmeter.testinfo.writer.TestInfoWriter;
@@ -637,6 +638,8 @@
 
 //                    setTestInfo(testInfo);
                     updateTestInfo();
+/*
+
                     if ((testInfo != null) & (testInfo.name != NEW) & (!testInfo.name.isEmpty()) &
                             (testInfoChecker == null || testInfoChecker.isInterrupted())) {
                         startTestInfoChecker();
@@ -644,16 +647,17 @@
                     if (testInfo.name.equals(NEW)) {
                         stopTestInfoChecker();
                     }
-/*
 
-                   if ((testInfo.name != NEW)) {
+*/
 
+                    if ((testInfo.name != NEW) & (!testInfo.name.isEmpty())) {
+                        TestInfoController.stop();
+                        TestInfoController.start(testInfo.id);
                     }
 
                     if (testInfo.name == NEW || (testInfo.name.isEmpty())) {
-
+                        TestInfoController.stop();
                     }
-*/
 
 
                 }
